backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/floatingnotification/BpkFloatingNotification.kt

```diff
@@ -0,0 +1,235 @@
+  }
+
+  val currentData = hostState.currentData
+  val accessibilityManager = LocalAccessibilityManager.current
+  LaunchedEffect(currentData) {
+    if (currentData != null) {
+      val duration = 4000L
+      delay(duration)
+      currentData.dismiss()
+    }
+  }
+
+  val data = hostState.currentData ?: return
+
+  Snackbar {
+    Box(
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(start = BpkSpacing.Base, end = BpkSpacing.Base, bottom = 30.dp),
+      contentAlignment = Alignment.BottomCenter
+    ) {
+      Box(
+        modifier = Modifier
+          .clip(RoundedCornerShape(BpkBorderRadius.Md))
+          .background(color = BpkTheme.colors.corePrimary)
+          .requiredSize(height = requiredSize.height, width = requiredSize.width)
+          .padding(start = BpkSpacing.Base, end = BpkSpacing.Base),
+        contentAlignment = Alignment.CenterStart
+      ) {
+        Row(
+          horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+          verticalAlignment = Alignment.CenterVertically
+        ) {
+          data.icon?.let { icon ->
+            BpkIcon(
+              icon = icon,
+              contentDescription = null,
+              size = BpkIconSize.Small,
+              tint = BpkTheme.colors.textOnDark
+            )
+          }
+          BpkText(
+            modifier = Modifier.weight(0.8f),
+            text = data.message,
+            maxLines = 2,
+            color = BpkTheme.colors.textOnDark,
+            style = BpkTheme.typography.footnote
+          )
+          data.action?.let { action ->
+            Box(
+              modifier = Modifier
+                .weight(0.2f)
+                .sizeIn(minHeight = BpkSpacing.Xxl, minWidth = BpkSpacing.Xxl)
+                .clickable { data.onClick?.invoke() },
+              contentAlignment = Alignment.Center
+            ) {
+              BpkText(
+                text = action,
+                textAlign = TextAlign.Center,
+                color = BpkTheme.colors.textOnDark,
+                style = BpkTheme.typography.label2
+              )
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+data class Cta(
+  val text: String,
+  val onClick: () -> Unit,
+)
+
+@Stable
+data class BpkFloatingNotificationData(
+  val message: String,
+  val action: String?,
+  val onClick: (() -> Unit)?,
+  val icon: BpkIcon?,
+  private val continuation: CancellableContinuation<SnackbarResult>
+)  {
+
+  fun performAction() {
+    if (continuation.isActive) continuation.resume(SnackbarResult.ActionPerformed, onCancellation = null)
+  }
+
+  fun dismiss() {
+    if (continuation.isActive) continuation.resume(SnackbarResult.Dismissed, onCancellation = null)
+  }
+
+}
+
+@Stable
+class BpkFloatingNotificationState(initiallyVisible: Boolean) {
+
+  private val mutex = Mutex()
+
+  var currentData by mutableStateOf<BpkFloatingNotificationData?>(null)
+    private set
+
+  suspend fun show(
+    message: String,
```

Comment:
sorry can we keep the naming for the props to keep consistency with figma/other platforms?

Reply:
changed them